### PR TITLE
DEPR: allow options for using bottleneck/numexpr

### PR DESCRIPTION
**Pull request by jreback**
deprecate pd.computation.expressions.set_use_numexpr()

supersedes #16140

### CODE REVIEW

File: pandas/core/config_init.py [hunk 1]
```diff
@@ -15,8 +15,40 @@
...
                                 is_one_of_factory, get_default_val,
                                 is_callable)
-from pandas.io.formats.format import detect_console_encoding
+from pandas.io.formats.console import detect_console_encoding
 
+# compute
+
+use_bottleneck_doc = """
+: bool
+    Use the bottleneck library to accelearte if its installed, default is True
```
Comment (crayxt) [inline pandas/core/config_init.py]: accelerate

**Comment by jreback** [comment]
cc @jorisvandenbossche 
cc @dhirschfeld 
cc @shoyer

File: doc/source/options.rst [hunk 2]
```diff
@@ -425,6 +425,10 @@ mode.use_inf_as_null                False        True means treat None, NaN, -IN
                                                  INF as null (old way), False means
                                                  None and NaN are null, but INF, -INF
                                                  are not null (new way).
+compute.use_bottleneck              True         Use the bottleneck library to accelerate
+                                                 computation if its installed
```
Comment (jorisvandenbossche) [inline doc/source/options.rst]: its -> it is

File: doc/source/options.rst [hunk 3]
```diff
@@ -425,6 +425,10 @@ mode.use_inf_as_null                False        True means treat None, NaN, -IN
...
+compute.use_numexpr                 True         Use the numexpr library to accelerate
+                                                 computation if its installed
```
Comment (jorisvandenbossche) [inline doc/source/options.rst]: same here

File: pandas/core/config_init.py (hunk 1 above)
Comment (jorisvandenbossche) [inline pandas/core/config_init.py]: same here and below about 'its'

**Comment by dhirschfeld** [comment]
LGTM

File: pandas/core/nanops.py [hunk 4]
```diff
@@ -23,9 +17,27 @@
...
+    _BOTTLENECK_INSTALLED = False
+
+_USE_BOTTLENECK = False
+
+
+def set_use_bottleneck(v=True):
+    # set/unset to use bottleneck
+    global _USE_BOTTLENECK
+    if _BOTTLENECK_INSTALLED:
+        _USE_BOTTLENECK = v
```
Comment (jorisvandenbossche) [inline pandas/core/nanops.py]: Should we warn here if ``v`` is False and bottleneck is not installed ?

**Review by jorisvandenbossche — COMMENTED** [review]
Can you add a whatsnew notice for the two new options?

**Comment by jreback** [comment]
- updated for comments
- added deprecation warning for ``union_categoricals``
- docs & tests for new ``compute.use*`` options

File: pandas/core/computation/expressions.py [hunk 5]
```diff
@@ -35,6 +36,9 @@ def set_use_numexpr(v=True):
     global _USE_NUMEXPR
     if _NUMEXPR_INSTALLED:
         _USE_NUMEXPR = v
+    elif v:
+        warnings.warn("trying to set the option to use numexpr, "
+                      "but it is not installed!")
```
Comment (jorisvandenbossche) [inline pandas/core/computation/expressions.py]: I know I suggested this, but, I think this one will be triggered by default if you don't have numexpr installed? (just by importing pandas, I mean, not only when setting the option manually when you don't have it installed)

Comment (jreback) [inline pandas/core/computation/expressions.py]: yep, just noticed that. going to take this out.

**Review by TomAugspurger — APPROVED** [review]

**Pull request merged**